synapse/storage/databases/main/keys.py

```diff
@@ -181,16 +181,18 @@ async def store_server_keys_json(
             desc="store_server_keys_json",
         )
 
-    async def get_server_keys_json(
+    async def get_server_keys_json_for_remote(
         self, server_keys: Iterable[Tuple[str, Optional[str], Optional[str]]]
     ) -> Dict[Tuple[str, Optional[str], Optional[str]], List[Dict[str, Any]]]:
         """Retrieve the key json for a list of server_keys and key ids.
         If no keys are found for a given server, key_id and source then
         that server, key_id, and source triplet entry will be an empty list.
         The JSON is returned as a byte array so that it can be efficiently
```

Comment:
En passant: I can't see any `bytes` in the return type. Is this comment outdated?

Reply:
I think it is just out of date? Will check.

Reply:
Yes, it is part of the `List[Dict[str, Any]]`; this is a list of the rows returned converted to dictionaries via `simple_select_list_txn`, which maps column name -> value. The value returned for the `key_json` will be `bytes` (or `memoryview`, I think).

Reply:
IMO this would be solved by not using `Any` (or...dare I say, stop auto-converting things to dictionaries of column name -> value since I suspect this is a performance footgun).

Reply:
Fine, that'll do for now. (I live in hope of a TypedDict/pydantic struct future)

Reply:
Yeah, I'd really like if it at least returned `TypedDict`... I don't think we can generate those dynamically from the helpers without introspecting the database. (AKA without an ORM.)